src/CurrencyNetwork.ts

```diff
@@ -1,75 +1,139 @@
+import * as ethUtils from 'ethereumjs-util'
+
 import { Utils } from './Utils'
 
-import * as ethUtils from 'ethereumjs-util'
+import {
+  Network,
+  NetworkDetails,
+  UserOverview,
+  UserOverviewRaw
+} from './typings'
 
+/**
+ * The CurrencyNetwork class contains all functions relevant for retrieving
+ * currency network related information.
+ */
 export class CurrencyNetwork {
+  private _utils: Utils
 
-  constructor (private utils: Utils) {
+  constructor (utils: Utils) {
+    this._utils = utils
   }
 
-  public getAll (): Promise<any[]> {
-    return this.utils.fetchUrl(`networks`)
+  /**
+   * Returns all registered currency networks.
+   */
+  public getAll (): Promise<Network[]> {
+    return this._utils.fetchUrl<Network[]>(`networks`)
   }
 
-  public getInfo (networkAddress: string): Promise<any> {
-    if (!this.utils.checkAddress(networkAddress)) {
-      return Promise.reject(`${networkAddress} is not a valid address.`)
+  /**
+   * Returns detailed information of specific currency network.
+   * @param networkAddress Address of a currency network.
+   */
+  public async getInfo (networkAddress: string): Promise<NetworkDetails> {
+    try {
+      await this._checkAddresses([networkAddress])
+      return this._utils.fetchUrl<NetworkDetails>(`networks/${networkAddress}`)
+    } catch (error) {
+      return Promise.reject(error)
     }
-    return this.utils.fetchUrl(`networks/${networkAddress}`)
   }
 
-  public getUsers (networkAddress: string): Promise<string[]> {
-    if (!this.utils.checkAddress(networkAddress)) {
-      return Promise.reject(`${networkAddress} is not a valid address.`)
+  /**
+   * Returns all addresses of users in a currency network.
+   * @param networkAddress Address of a currency network.
+   */
+  public async getUsers (networkAddress: string): Promise<string[]> {
+    try {
+      await this._checkAddresses([networkAddress])
+      return this._utils.fetchUrl<string[]>(`networks/${networkAddress}/users`)
+    } catch (error) {
+      return Promise.reject(error)
     }
-    return this.utils.fetchUrl(`networks/${networkAddress}/users`)
   }
 
-  public getUserOverview (networkAddress: string, userAddress: string): Promise<any> {
-    if (!this.utils.checkAddress(networkAddress)) {
-      return Promise.reject(`${networkAddress} is not a valid address.`)
+  /**
+   * Returns overview of an user in a specific currency network.
```

Comment:
its 'a user' 
repeats through this document.

Reply:
fixed